lib/segment/src/payload_storage/condition_checker.rs

```diff
@@ -150,6 +155,23 @@ impl ValueChecker for GeoRadius {
     }
 }
 
+impl ValueChecker for GeoPolygon {
+    fn check_match(&self, payload: &Value) -> bool {
+        match payload {
+            Value::Object(obj) => {
+                let lon_op = obj.get("lon").and_then(|x| x.as_f64());
+                let lat_op = obj.get("lat").and_then(|x| x.as_f64());
+
+                if let (Some(lon), Some(lat)) = (lon_op, lat_op) {
+                    return self.check_point(lon, lat);
+                }
+                false
+            }
+            _ => false,
+        }
+    }
```

Comment:
Since this is the third time this code is repeated, we can abstract it away. I will make a suggestion on a separate PR

Reply:
https://github.com/zzzz-vincent/qdrant/pull/1, as promised

Reply:
ack. Will do in the next PR.